packages/core/src/StateMachine.ts

```diff
@@ -193,6 +194,19 @@ export class StateMachine<
 
     this.states = this.root.states; // TODO: remove!
     this.events = this.root.events;
+
+    if (isDevelopment) {
+      if (
+        Object.values(this.states).some(
+          (state) => state.type === 'final' && !!state.output
+        ) &&
+        !this.root.output
```

Comment:
I'd flip those conditions - first check the `!this.root.output` thing, then do the `Object.values` work.

It's a nit though since it doesn't matter here